include/tscore/SourceLocation.h

```diff
@@ -34,9 +34,9 @@
 class SourceLocation
 {
 public:
-  const char *file = nullptr;
-  const char *func = nullptr;
-  int line         = 0;
+  const char *file;
+  const char *func;
+  int line;
 
   SourceLocation()                          = default;
   SourceLocation(const SourceLocation &rhs) = default;
```

Comment:
The following indicates we can probably just eliminate the default constructor:
```
wkaras ~/REPOS/TS
O$ grep '[^e]SourceLocation[^,]*[({]' $(findsrc) # | fgrep -v 'MakeSourceLocation('
./include/tscore/Regression.h:#define RegressionMakeLocation(f) SourceLocation(__FILE__, f, __LINE__)
./include/tscore/SourceLocation.h:#define MakeSourceLocation() SourceLocation(__FILE__, __FUNCTION__, __LINE__)
./include/tscore/SourceLocation.h:  SourceLocation()                          = default;
./include/tscore/SourceLocation.h:  SourceLocation(const SourceLocation &rhs) = default;
./include/tscore/SourceLocation.h:  SourceLocation(const char *_file, const char *_func, int _line) : file(_file), func(_func), line(_line) {}
./include/tscore/Diags.h:    static const SourceLocation DiagsError_loc = MakeSourceLocation();  \
./include/tscore/Diags.h:    static const SourceLocation STDE_loc = MakeSourceLocation(); \
./include/tscore/Diags.h:    static const SourceLocation DiagsErrorV_loc = MakeSourceLocation();  \
./include/tscore/Diags.h:    static const SourceLocation STDEV_loc = MakeSourceLocation(); \
./include/tscore/Diags.h:      static const SourceLocation Diag_loc = MakeSourceLocation(); \
./include/tscore/Diags.h:    static const SourceLocation DbgPrintf_loc = MakeSourceLocation();                  \
./iocore/eventsystem/I_Lock.h:      m->srcloc  = SourceLocation(nullptr, nullptr, 0);
wkaras ~/REPOS/TS
O$ grep '^SourceLocation[^,]*[({]' $(findsrc) # | fgrep -v 'MakeSourceLocation('
./src/tscore/SourceLocation.cc:SourceLocation::str(char *buf, int buflen) const
./src/tscore/SourceLocation.cc:SourceLocation::print(ts::BufferWriter &w, ts::BWFSpec const &) const
wkaras ~/REPOS/TS
O$ 
```
We can probably also change the data members to:
```
  const char * const file;
  const char * const func;
  const int line;
```

Reply:
Unfortunately those causes compile errors.

Reply:
You're right, sorry.